src/viam/app/data_client.py

```diff
@@ -1696,6 +1816,156 @@ async def _file_upload(self, metadata: UploadMetadata, file_contents: FileData)
                 raise TypeError("Response cannot be empty")
             return response
 
+    async def get_data_pipeline(self, id: str) -> DataPipeline:
+        """Get a data pipeline by its ID.
+
+        ::
+
+            data_pipeline = await data_client.get_data_pipeline(id="<YOUR-DATA-PIPELINE-ID>")
+
+        Args:
+            id (str): The ID of the data pipeline to get.
+
+        Returns:
+            DataPipeline: The data pipeline with the given ID.
+        """
+        request = GetDataPipelineRequest(id=id)
+        response: GetDataPipelineResponse = await self._data_pipelines_client.GetDataPipeline(request, metadata=self._metadata)
+        return self._proto_pipeline_to_pipeline(response.data_pipeline)
+
+    async def list_data_pipelines(self, organization_id: str) -> List[DataPipeline]:
+        """List all of the data pipelines for an organization.
+
+        ::
+
+            data_pipelines = await data_client.list_data_pipelines(organization_id="<YOUR-ORGANIZATION-ID>")
+
+        Args:
+            organization_id (str): The ID of the organization that owns the pipelines.
+                You can obtain your organization ID from the Viam app's organization settings page.
+
+        Returns:
+            List[DataPipeline]: A list of all of the data pipelines for the given organization.
+        """
+        request = ListDataPipelinesRequest(organization_id=organization_id)
+        response: ListDataPipelinesResponse = await self._data_pipelines_client.ListDataPipelines(request, metadata=self._metadata)
+        return [self._proto_pipeline_to_pipeline(pipeline) for pipeline in response.data_pipelines]
```

Comment:
All of these would change to use `DataPipeline.from_proto` once you implement that